examples/events/events.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/nlopes/slack"
+)
+
+// You more than likely want your "Bot User OAuth Access Token" which starts with "xoxb-"
+var api = slack.New("TOKEN")
+
+func main() {
+	http.HandleFunc("/events-endpoint", func(w http.ResponseWriter, r *http.Request) {
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(r.Body)
+		body := buf.String()
+		eventsAPIEvent, e := api.ParseEventsAPIEvent(json.RawMessage(body))
+		if e != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+
+		if eventsAPIEvent.Type == slack.URLVerification {
+			var r *slack.ChallengeResponse
+			err := json.Unmarshal([]byte(body), &r)
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+			}
+			w.Header().Set("Content-Type", "text")
+			w.Write([]byte(r.Challenge))
+		}
+		if eventsAPIEvent.Type == slack.CallbackEvent {
+			postParams := slack.PostMessageParameters{}
+			innerEvent := eventsAPIEvent.InnerEvent
+			switch ev := innerEvent.Data.(type) {
+			case *slack.AppMentionEvent:
+				api.PostMessage(ev.Channel, "Yes, hello.", postParams)
```

Comment:
Two notes on this example:

1. You need to write back an OK, otherwise slack will redeliver the event
2. Sending a reply should probably be done after writing OK in case slack's API times out. Slack will redeliver if you don't respond within 3 seconds. https://api.slack.com/events-api#graceful_retries